src/test/kotlin/no/nav/amt/distribusjon/journalforing/JournalforingServiceTest.kt

```diff
@@ -1,100 +1,60 @@
 package no.nav.amt.distribusjon.journalforing
 
 import io.kotest.matchers.shouldBe
-import io.mockk.clearMocks
-import io.mockk.coEvery
-import io.mockk.coVerify
-import io.mockk.mockk
+import io.kotest.matchers.shouldNotBe
 import kotlinx.coroutines.runBlocking
-import no.nav.amt.distribusjon.hendelse.model.HendelseType
-import no.nav.amt.distribusjon.journalforing.dokarkiv.DokarkivClient
+import kotlinx.coroutines.time.delay
+import no.nav.amt.distribusjon.Environment
+import no.nav.amt.distribusjon.application.plugins.objectMapper
+import no.nav.amt.distribusjon.distribusjonskanal.Distribusjonskanal
+import no.nav.amt.distribusjon.hendelse.model.HendelseDto
+import no.nav.amt.distribusjon.integrationTest
 import no.nav.amt.distribusjon.journalforing.model.Journalforingstatus
-import no.nav.amt.distribusjon.journalforing.pdf.PdfgenClient
-import no.nav.amt.distribusjon.journalforing.person.AmtPersonClient
-import no.nav.amt.distribusjon.journalforing.sak.SakClient
-import no.nav.amt.distribusjon.utils.SingletonPostgresContainer
-import no.nav.amt.distribusjon.utils.TestRepository
+import no.nav.amt.distribusjon.utils.AsyncUtils
+import no.nav.amt.distribusjon.utils.MockResponseHandler
 import no.nav.amt.distribusjon.utils.data.HendelseTypeData
 import no.nav.amt.distribusjon.utils.data.Hendelsesdata
-import no.nav.amt.distribusjon.utils.data.Journalforingdata
 import no.nav.amt.distribusjon.utils.data.Persondata
+import no.nav.amt.distribusjon.utils.produceStringString
+import org.apache.kafka.clients.producer.ProducerRecord
 import org.junit.Assert.assertThrows
-import org.junit.Before
-import org.junit.BeforeClass
 import org.junit.Test
-import java.time.LocalDate
+import java.time.Duration
 import java.time.LocalDateTime
 
 class JournalforingServiceTest {
```

Comment:
Jeg synes kanskje vi taper litt på å gjøre akkurat denne testen til integrasjonstest i stedet for å bruke mockene. Fordelen med integrasjonstest er jo selvsagt at vi får testet helt ut i klientene, og det er jo bra, men med mockene kunne vi teste at det vi sender til klientene er riktig (altså at vi bruker riktige data). Man kan alternativt splitte ut funksjonalitet som går på å lage de riktige requestene i public-metoder og lage egne tester for dem, for da slipper man å sjekke på hva som sendes til mockene. De testene som sjekker at ingenting blir endret føler jeg også at er litt sårbare, for de kan jo gå grønt hvis oppsettet bare ikke funker. I tillegg tar integrasjonstestene lengre tid å kjøre. Det er ikke veldig viktig for meg, men akkurat her synes jeg mockene ga mer nytte.

Reply:
(jeg er veldig glad i integrasjonstester altså, så jeg blir ikke sur eller noe hvis du lar den være sånn 😊 Hvis vi kjenner behov for å teste at det vi skal sende til klientene blir riktig kan jo det testes på andre måter)

Reply:
Grunnen til at jeg ikke ønsker å bruke mocker er hovedsaklig at de er, for meg, veldig kjedelige å vedlikeholde, da en hver endring ofte krever at alle testene må endres. Vi kan jo gjøre om alle disse mock-clientene i integrasjonstestene til å være `mockk` i stede for noe hjemmesnekra, det er ikke så viktig å teste clientene i seg selv. Men jeg vet ikke om jeg synes at det å teste på hvilke argumenter vi kaller en metode med er særlig bra, det er kanskje heller et tegn på at koden ikke er velldig testbar. Jeg synes det ble løst for dataene til pdfen med å trekke ut logikken til en egen funksjon, og tenker jo at det er litt bedre enn å mocke opp alt. Vi kan gjøre det samme med `getJournalpostRequest` også men jeg vet ikke helt om jeg skjønner hva som er viktig å teste her.

De testene som sjekker at ingenting blir endret er jeg helt enig i at ikke er spessielt gode, så jeg har skrevet om de hvor det ikke er nødvendig å vente for å se hva som skjer, så de går litt raskere og er mer pålitelig. Men det er vanskelig å teste at ikke noe skjer, så det er kanskje et argument for å mocke, kanskje det finnes en mellomting?

Jeg vet ikke om du har prøvd å skru på reusable containers? Hos meg tar alle testene 6 sekunder inkludert bygg, så jeg føler ikke at jeg sitter å venter mye på de her.